src/attributes.jl

```diff
@@ -180,6 +180,21 @@ function h5readattr(filename, name::AbstractString)
     dat
 end
 
+"""
+    num_attrs()
+
+Retrieve the number of attributes from an object.
+
+See [`API.h5o_get_info`](@ref).
+"""
+function num_attrs(obj)
+    info = @static if API._libhdf5_build_ver < v"1.12.0"
```

Comment:
Out of curiosity does using `num_attrs` provide any speed benefits vs.  the existing check on `object_info`?

Reply:
Read through this forum thread:
https://forum.hdfgroup.org/t/h5oget-info-performance-with-large-number-of-chunks/2093/6

Basically `H5Oget_info` pulls a lot of information that we do not need which can be quite slow in certain circumstances. They resolved this by changing to the API so you can ask exactly for the data that you need, in this case the number of attributes.

Reply:
Makes sense

Reply:
```
julia> @btime API.h5o_get_info1(h5f["lore"]).num_attrs
  5.840 μs (5 allocations: 416 bytes)
0x0000000000000001

julia> @btime API.h5o_get_info(h5f["lore"], API.H5O_INFO_NUM_ATTRS).num_attrs
  4.757 μs (5 allocations: 224 bytes)
0x0000000000000001

```
Defintely allocates a lot less even with small info structs.